### PR TITLE
Add ptr and enums

### PR DESCRIPTION
**Pull request by jlobraco**
Added native pointer types
Added enums

### CODE REVIEW

**Comment by pkuprys** [comment]
What do we need these for? Is this getting us up to date with the latest from the fork? Or us adding some custom stuff?

**Comment by jlobraco** [comment]
> What do we need these for? Is this getting us up to date with the latest from the fork? Or us adding some custom stuff?

I needed to regenerate the datatrain messages, and since the last time I did this, they added stuff that wasn't supported by the generator. Namely unions of native types and null (i.e. pointers to native types), and enums.

File: gen_codecs.go [hunk 1]
```diff
@@ -107,30 +128,31 @@ func NewUnionCodecGenerator(codecFromIndex []*Codec) (*CodecGenerator, error) {
...
 
 	gen.genDecodeInstanceSrc = func() string {
 		var w bytes.Buffer
 
-		w.WriteString(fmt.Sprintf("func(buf []byte) (%s, []byte, error) {\n", gen.genNativeTypeNameSrc()))
+		w.WriteString(fmt.Sprintf("func(buf []byte) (%s, []byte, error) {\n", gen.genNativeTypeNamePtrSrc()))
 		w.WriteString("tmpBuf := buf\n")
 		w.WriteString("idx, tmpBuf, err := goavro.LongNativeFromBinary(tmpBuf)\n")
-		w.WriteString(fmt.Sprintf("if err != nil { return %s, buf, err }\n", gen.genNativeDefaultValueSrc()))
+		w.WriteString(fmt.Sprintf("if err != nil { return %s, buf, err }\n", "nil"))
```
Comment (pkuprys) [inline gen_codecs.go]: It seems like `Sprintf` can be removed.
```suggestion
		w.WriteString("if err != nil { return nil, buf, err }\n")
```

File: gen_codecs.go [hunk 2]
```diff
@@ -107,30 +128,31 @@ func NewUnionCodecGenerator(codecFromIndex []*Codec) (*CodecGenerator, error) {
...
 		w.WriteString("switch idx {\n")
 
 		for i, fieldCodec := range codecFromIndex {
 			w.WriteString(fmt.Sprintf("case %d:\n", i))
 			if codecFromIndex[i].typeName.fullName == "null" {
 				w.WriteString("// Null case, use empty value\n")
-				w.WriteString(fmt.Sprintf("return %s, tmpBuf, nil\n", gen.genNativeDefaultValueSrc()))
+				w.WriteString(fmt.Sprintf("return %s, tmpBuf, nil\n", "nil"))
```
Comment (pkuprys) [inline gen_codecs.go]: Here too:
```suggestion
				w.WriteString("return nil, tmpBuf, nil\n")
```

File: gen_codecs.go [hunk 3]
```diff
@@ -107,30 +128,31 @@ func NewUnionCodecGenerator(codecFromIndex []*Codec) (*CodecGenerator, error) {
...
 			} else {
-				w.WriteString(fmt.Sprintf("return  %s(tmpBuf)\n", fieldCodec.generator.genDecodeInstanceSrc()))
+				w.WriteString(fmt.Sprintf("return  %s(tmpBuf)\n", fieldCodec.generator.genDecodePtrInstanceSrc()))
 			}
 		}
 		w.WriteString(fmt.Sprintf("default:\n"))
-		w.WriteString(fmt.Sprintf("return %s, buf, fmt.Errorf(\"union index out of bounds\")\n", gen.genNativeDefaultValueSrc()))
+		w.WriteString(fmt.Sprintf("return %s, buf, fmt.Errorf(\"union index out of bounds\")\n", "nil"))
```
Comment (pkuprys) [inline gen_codecs.go]: and here
```suggestion
		w.WriteString("return nil, buf, fmt.Errorf(\"union index out of bounds\")\n")
```

**Review by pkuprys — CHANGES_REQUESTED** [review]
small opportunity for cleanup

File: gen_codecs.go (hunk 1 above)
Comment (jlobraco) [reply gen_codecs.go]: Yea, just me being lazy :)

I'll make the changes

Comment (jlobraco) [reply gen_codecs.go]: done

File: gen_codecs.go (hunk 2 above)
Comment (jlobraco) [reply gen_codecs.go]: done

File: gen_codecs.go (hunk 3 above)
Comment (jlobraco) [reply gen_codecs.go]: done

**Review by pkuprys — APPROVED** [review]

**Pull request merged**